services/web/server/src/simcore_service_webserver/garbage_collector/_tasks_trash.py

```diff
@@ -0,0 +1,66 @@
+"""
+    Scheduled tasks addressing users
+
+"""
+
+import asyncio
+import logging
+from collections.abc import AsyncIterator, Callable
+
+from aiohttp import web
+from tenacity import retry
+from tenacity.before_sleep import before_sleep_log
+from tenacity.wait import wait_exponential
+
+from ..projects._trash_api import prune_all_trashes
+
+_logger = logging.getLogger(__name__)
+
+CleanupContextFunc = Callable[[web.Application], AsyncIterator[None]]
+
+
+_PERIODIC_TASK_NAME = f"{__name__}"
+_APP_TASK_KEY = f"{_PERIODIC_TASK_NAME}.task"
+
+
+@retry(
+    wait=wait_exponential(min=5, max=20),
+    before_sleep=before_sleep_log(_logger, logging.WARNING),
+)
+async def _run_task(app: web.Application):
+    if deleted := await prune_all_trashes(app):
+        for name in deleted:
+            _logger.info("Trash item %s expired and was deleted", f"{name}")
+    else:
+        _logger.info("No trash items expired")
+
+
+async def _run_periodically(app: web.Application, wait_interval_s: float):
+    while True:
+        await _run_task(app)
+        await asyncio.sleep(wait_interval_s)
+
+
+def create_background_task_to_prune_trash(
```

Comment:
Q: why are you re-creating `packages/service-library/src/servicelib/background_task.py` ? it works also without Redis.

Reply:
This is an old code. I just decided to duplicate what was there for this first iteration. I also created https://github.com/ITISFoundation/osparc-simcore/issues/6592 to follow up.

Let me adapt **all** of them right away in a separate PR.